### DISCUSSION TITLE
'Temporary' fix for logid not functioning after boot

### DISCUSSION THREAD

**Discussion by Gusfit1** [PixlOne/logiops#480] [General]
With EndeavourOS my system seems to have an issue running logid on boot. `sudo systemctl status logid` returns with 'Active: active (running)', but is non-functional. It seems it relies on another service to be running first and gets loaded at the wrong time. An easy 'temporary' fix for this is to create a timer for the logid service:

/usr/lib/systemd/system/logid.timer
```
[Unit]
Description=Run logid on boot

[Timer]
OnBootSec=10

[Install]
WantedBy=timers.target
```
- Disable logid.service - `sudo systemctl disable logid.service`
- Enable logid.timer - `sudo systemctl disable logid.timer`
- Reboot (possibly need to shut down to confirm timer is set properly for a cold boot)

Now logid.timer will be called and start logid.service 10 seconds later. Note that you might need to edit the timer to suit your own system, for example `1min` is much too long of a delay and `5`(s) is too quick for my system. 

A better fix would be to check that the required services are running before starting logid, but that's above my paygrade.

More info on timers: https://wiki.archlinux.org/title/Systemd/Timers

**Comment by Grummel1** [comment]
"It seems it relies on another service to be running first"

Do you know which service? I set up my first linux Mint and logid is active (running) But it is non functional. Even after reboot. 
` logid.service - Logitech Configuration Daemon
     Loaded: loaded (/usr/lib/systemd/system/logid.service; disabled; preset: e>
     Active: active (running) since Sun 2025-05-18 19:39:17 CEST; 29min ago
   Main PID: 1702 (logid)
      Tasks: 9 (limit: 76409)
     Memory: 2.9M (peak: 4.8M)
        CPU: 60ms
     CGroup: /system.slice/logid.service
             └─1702 /usr/local/bin/logid
`
